src/lib/optimizer/strategy/predicate_merge_rule.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+
+namespace opossum {
+
+/**
+ * This rule cleans up any large subplan that the PredicateSplitUpRule created after other optimizer rules have run.
```

Comment:
Suggestion "The PredicateSplitUpRule might have rewritten disjunctions into complex predicate/union chains. If no rule between the PredicateSplitUpRule and this rule has benefited from these chains, executing them as predicates and unions might be more expensive than having the ExpressionEvaluator run on the original monolithic expression. Controlled by [varname, see below], this rule reverts the PredicateSplitUpRule's changes by merging multiple PredicateNodes and UnionNodes into single a PredicateNode"?